### PR TITLE
add home work 14

### PR DESCRIPTION
**Pull request by NataliiaFed**
home work #14

### CODE REVIEW

File: phonebook.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            key_list = phonebook.keys()
+            print(len(key_list))
+
+        case 'add': # add: додати запис
+            name = input("Enter a name: ")
+
+            if name in phonebook:
+                print("Contact with this name already exists.")
+            else:
+                while True:
```
Comment (klymenok) [inline phonebook.py:44]: If you want to use three separate patters and check them separately it is not necessary to put them in the infinite loop or use such a big construction of `if .. elif ... else`. You can simplify your code like this (example where the infinite loop is still using):
```
if any([re.fullmatch(r'\+380\d{9}$', phonenumber), re.fullmatch(r'380\d{9}(?=$)', phonenumber), re.fullmatch(r'0\d{9}\b', phonenumber)]):
    break
else:
    print("Invalid format of a phone number. Try again.")

File: email_finder.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import re
+
+text_file = input("File name: ")
+with open(text_file, 'r') as file:
+    text = file.read()
+
+    print(re.findall(r'[a-zA-Z0-9_.]+@(?:[a-zA-Z0-9-]+\.)+[a-zA-z]{2,}', text)) # any other symbols relevant for email address can be added in []
```
Comment (klymenok) [inline email_finder.py]: The task was to replace emails. You shouldn't print them all

File: email_finder.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    new_text = re.sub(r'[a-zA-Z0-9_.]+@(?:[a-zA-Z0-9-]+\.)+[a-zA-z]{2,}', '***@***', text)
+    print(new_text)
+
+    new_text2 = text
+    emails_iter = re.finditer(r'[a-zA-Z0-9_.]+@(?:[a-zA-Z0-9-]+\.)+[a-zA-z]{2,}', new_text2)
+    for item in emails_iter:
+        email_to_replace = item.group()[1:-1]
+        print(email_to_replace)
```
Comment (klymenok) [inline email_finder.py]: Don't print real emails in this task

**Review by klymenok — CHANGES_REQUESTED** [review]

File: phonebook.py (hunk 1 above)
Comment (NataliiaFed) [reply phonebook.py:44]: Created one pattern for all cases

File: email_finder.py (hunk 2 above)
Comment (NataliiaFed) [reply email_finder.py]: Just forgot to remove print lines from the code before submitting (used it to test the code)

**Review by klymenok — APPROVED** [review]
Looks good!

**Pull request merged**